qiskit_addon_dice_solver/dice_solver.py

```diff
@@ -491,22 +491,25 @@ def _construct_ci_vec_from_amplitudes(
     amps: list[float], ci_strs: list[list[int]]
 ) -> tuple[np.ndarray, np.ndarray, np.ndarray]:
     """Construct wavefunction amplitudes from CI strings and their associated amplitudes."""
-    uniques = np.unique(np.array(ci_strs))
-    num_dets = len(uniques)
-    ci_vec = np.zeros((num_dets, num_dets))
-    ci_strs_a = np.zeros(num_dets, dtype=np.int64)
-    ci_strs_b = np.zeros(num_dets, dtype=np.int64)
+    uniques_a = np.unique(np.array([s[0] for s in ci_strs]))
+    uniques_b = np.unique(np.array([s[1] for s in ci_strs]))
```

Comment:
```suggestion
    strs_a, strs_b = zip(*ci_strs)
    uniques_a = np.unique(strs_a)
    uniques_b = np.unique(strs_b)
```
I think this should work.